flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveTableSource.java

```diff
@@ -100,10 +110,41 @@ private HiveTableSource(JobConf jobConf, ObjectPath tablePath, CatalogTable cata
 	}
 
 	@Override
-	public InputFormat getInputFormat() {
+	public boolean isBounded() {
+		return true;
+	}
+
+	@Override
+	public DataStream<Row> getDataStream(StreamExecutionEnvironment execEnv) {
 		if (!initAllPartitions) {
 			initAllPartitions();
 		}
+		@SuppressWarnings("unchecked")
+		TypeInformation<Row> typeInfo = (TypeInformation<Row>) fromDataTypeToTypeInfo(getProducedDataType());
+		HiveTableInputFormat inputFormat = getInputFormat();
+		DataStreamSource<Row> source = execEnv.createInput(inputFormat, typeInfo);
+
+		Configuration conf = GlobalConfiguration.loadConfiguration();
+		if (conf.getBoolean(TABLE_EXEC_HIVE_INFER_SOURCE_PARALLELISM)) {
+			int max = conf.getInteger(TABLE_EXEC_HIVE_INFER_SOURCE_PARALLELISM_MAX);
```

Comment:
validate user configured max value

Reply:
It is a config option, it has default value. Don't need force user to configure it.

Reply:
I mean do some validation to make sure user configured a valid value.